nsecx.go

```diff
@@ -93,3 +93,62 @@ func (rr *NSEC3) Match(name string) bool {
 	}
 	return false
 }
+
+// compares domains according to the canonical ordering specified in RFC4034
+// returns an integer value similar to strcmp
+// (0 for equal values, -1 if s1 < s2, 1 if s1 > s2)
+func DomainCompare(s1, s2 string) int {
```

Comment:
we already have CompareDomainName, so this probably needs to be renamed, although I don't know into what, mayby `Less` - from the `sort` package? Also the function signature needs to be different as well, so it can fit that pkg.

It's also very allocation heavy, so that should be fixed, and the use of underscores a not typical Go, so that needs fixing as well.

Comparing names in this fashion

Reply:
Hopefully resolved by the new commit.